### DISCUSSION TITLE
Toggle multiple comments simultaneously

### DISCUSSION THREAD

**Discussion by donjan** [zed-industries/zed#17876] [Feature Requests]
I'm quite used to the workflow of toggling on and off multiple commented lines simultaneously with one hotkey. A video speaks a thousand words:

https://github.com/user-attachments/assets/f3c508af-e6b2-4ee4-b871-7b997c974489

Is there a way of achieving this in Zed, or do we have to wait for an extension / core support? Currently Zed double-comments the already commented lines.

**Comment by omarjatoi** [comment]
I wondered this too, but it does not seem like this is supported. Looking at the editor "toggle comments" action:
https://github.com/zed-industries/zed/blob/e3d54b22118f54d13488707bc3cb9f0e9143b1c8/crates/editor/src/actions.rs#L79-L83

There could be an additional option added to support something like "multi-line toggle" or something similar, which checks the selected lines first, removing or adding the comment syntax as needed based on current line state.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
